lib/Db/MessageMapper.php

```diff
@@ -501,6 +503,7 @@ public function updatePreviewDataBulk(Message ...$messages): array {
 					$message->getPreviewText() === null ? null : mb_strcut($message->getPreviewText(), 0, 255),
```

Comment:
This email [mail_5322.txt](https://github.com/nextcloud/mail/files/9354885/mail_5322.txt) uses a weird encoding. It breaks the updatePreviewDataBulk process for me and rollback all changes. 

The following code made it work. I'm not 100% sure if that's not a problem with my dev setup somehow. 
```
$previewText = $message->getPreviewText();
if ($previewText !== null) {
    $previewText = mb_strcut(mb_convert_encoding($previewText, 'UTF-8', 'UTF-8'), 0, 255);
}
```

Reply:
I think this is a general issue with the encoding that we've been having for a while now, see #5834 or #6913 for example.